jvm/src/main/kotlin/app/cash/trifle/Certificate.kt

```diff
@@ -86,13 +71,13 @@ data class Certificate internal constructor(
         if (certificateRequest.pkcs10Req.subject == x509Certificate.subject
           && certificateRequest.pkcs10Req.subjectPublicKeyInfo == x509Certificate.subjectPublicKeyInfo
         ) {
-          VerifyResult(SUCCESS)
+          return true
         } else {
-          VerifyResult(CSR_MISMATCH)
+          throw CSRMismatchException("Trifle certificate does not match CSR")
         }
       }
     }
-    return VerifyResult(UNSPECIFIED_FAILURE)
+    throw UnSpecifiedFailureException("Unspecified Trifle verification failure")
```

Comment:
This is an unnecessary thrown exception here. The nice thing about sealed classes is that all the subclasses can be resolved at runtime, so this is technically unreachable code.

Reply:
good catch!